raiden/raiden_service.py

```diff
@@ -823,3 +828,18 @@ def target_mediated_transfer(self, transfer: LockedTransfer):
         self.start_health_check_for(transfer.initiator)
         init_target_statechange = target_init(transfer)
         self.handle_state_change(init_target_statechange)
+
+    def upgrade_db(self, current_version: int, old_version: int):
+        log.debug(f'Upgrading database from v{old_version} to v{current_version}')
+        # Prevent unique constraint error in DB when recording raiden "runs"
+        gevent.sleep(1)
```

Comment:
Everytime the storage is instantiated, it writes a `run log`:
https://github.com/raiden-network/raiden/blob/2e15af11886337a62e3ef17a80eee85927f3636d/raiden/storage/sqlite.py#L53

Where the the time is a primary key value (should be unique):
https://github.com/raiden-network/raiden/blob/2e15af11886337a62e3ef17a80eee85927f3636d/raiden/storage/utils.py#L45

Therefore, creating a 1 second delay prevents unique constraint failures from happening

Reply:
Okay, so we do have two instances of the storage, one from `RaidenService.start` and another from `UpgradeManager.run`. I think we should try to have only one, to avoid the problems with multiple connections which might corrupt the database, what about:

- removing the call `_log_raiden_run` from the `SQLiteStorage` constructor
- run the upgrade through the `UpgradeManager`, and instantiate it first (before the `SQLiteStorage` )
- after upgrade manager executed, then instantiate the `SQLiteStorage` and run `_log_raiden_run`

This would remove the artificial sleeps, and make sure we have only one connection to the latest db